brainglobe_atlasapi/atlas_generation/atlas_scripts/allen_bbp_ccfv3a_entire_mouse_brain.py

```diff
@@ -0,0 +1,267 @@
+import json
+import zipfile
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Tuple, Union
+
+import nrrd
+import numpy as np
+import requests
+from brainglobe_utils.IO.image import load_any
+from scipy.ndimage import zoom
+from tqdm import tqdm
+
+from brainglobe_atlasapi.atlas_generation.mesh_utils import (
+    construct_meshes_from_annotation,
+)
+from brainglobe_atlasapi.atlas_generation.wrapup import wrapup_atlas_from_data
+
+# Copy-paste this script into a new file and fill in the functions to package
+# your own atlas.
+
+### Metadata ###
+
+# The minor version of the atlas in the brainglobe_atlasapi, this is internal,
+# if this is the first time this atlas has been added the value should be 0
+# (minor version is the first number after the decimal point, ie the minor
+# version of 1.2 is 2)
+
+
+@dataclass
+class AtlasMetadata:
+    """
+    Holds metadata describing a BrainGlobe atlas.
+
+    Attributes:
+        version (int): The minor version of the atlas (the first number after
+            the decimal point).
+        name (str): Atlas name, following "FirstAuthor_SpeciesCommonName" or
+            "Institution_SpeciesCommonName".
+        citation (str): A DOI of the most relevant citable document.
+        species (str): The scientific name of the species.
+        atlas_link (Union[str, Tuple[str, ...]]): URL(s) for the data files.
+        orientation (str): The **original** atlas orientation in BrainGlobe
+            convention.
+        root_id (int): The ID of the highest atlas level. This is commonly
+            called root or brain.
+        resolution (Union[int, float]): The atlas resolution in microns.
+    """
+
+    version: int
+    name: str
+    citation: str
+    species: str
+    atlas_link: Union[str, Tuple[str, ...]]
+    orientation: str
+    root_id: int
+    resolution: Union[int, float]
+
+
+METADATA = AtlasMetadata(
+    version=0,
+    name="allen_bbp_ccfv3a_entire_mouse_brain",
+    citation="https://doi.org/10.1101/2024.11.06.622212",
+    atlas_link="https://zenodo.org/api/records/14034334/files-archive",
+    species="Mus musculus",
+    orientation="spr",
+    root_id=997,
+    resolution=25,
+)
```

Comment:
Not a strong opinion, but maybe a worry would be that using a dataclass (while nice syntactic sugar for more experienced developers) introduces a small barrier for novice programmers to contribute atlases in the future (because they need to understand what `e.g. METADATA.name` refers to)?

Reply:
Interesting point, I don't think its so difficult for novices but its hard to say. I think the main point of this change is to organise all the metadata in one place with detailed descriptions of what each field requires and what type of data it expects. I think doing this with constants and comments in a template file is less than ideal perhaps but I'm open to opinions.

Reply:
yea, it's a trade-off. Not sure - will bring up at dev meeting tomorrow.

Reply:
I would vote against dataclasses. Atlas contributors tend to be the least experienced contributors we have (some have no Python experience). Anything we can do to simplify this process is good.

Reply:
fair point! should I remove it from this script or is it ok

Reply:
Let's remove it for consistency across packaging scripts?